src/fitnesse/wikitext/parser/SymbolTypeDecorator.java

```diff
@@ -1,18 +0,0 @@
-package fitnesse.wikitext.parser;
-
-public abstract class SymbolTypeDecorator extends SymbolType implements Translation{
```

Comment:
I see your point. Do you know of existing subclasses? Maybe the author of the class could tell whether he's extended this in a plugin?

In general, I believe the old `SymbolTypeDecorator` was an unfortunate API for several reasons, the most important being that it is basically impossible to "decorate" an existing `Translation`. You can override it completely or not at all. As a result, the `ColoredSlimTable` duplicated translation logic of the `Table`.

I'll think of a way to preserve backwards compatibility.
However, I suggest to try removing this feature completely (with documentation of eventual migration steps).

Reply:
I was thinking about this a little bit more and I think there is minimal to no risk in removing the `SymbolTypeDecorator` class...

To use it, you wrap another symbol type and either
1. assign it the original symbol type name (to override the original symbol type, as with `ColoredSlimTable`), or
2. assign it a new name (to create a new symbol type reusing the base type's logic in a very strange way)

If we accept that scenario (2) doesn't make sense and can be ignored then I argue that only instance of scenario (1) was the `ColoredSlimTable`. This is because the only way to use it is inside the Fitnesse project itself. Plugins cannot create and inject an overridden symbol type - matching the same symbol type name - because there is is no programmatic way of removing the original symbol type and thus `SymbolProvider` would always use the original one.